include/pybind11/chrono.h

```diff
@@ -194,17 +194,17 @@ class type_caster<std::chrono::time_point<std::chrono::system_clock, Duration>>
         std::time_t tt
             = system_clock::to_time_t(time_point_cast<system_clock::duration>(src - us));
 
-        std::tm localtime;
-        std::tm *localtime_ptr = localtime_thread_safe(&tt, &localtime);
-        if (!localtime_ptr) {
-            throw cast_error("Unable to represent system_clock in local time");
+        std::tm gmtime;
+        std::tm *gmtime_ptr = gmtime_thread_safe(&tt, &gmtime);
+        if (!gmtime_ptr) {
+            throw cast_error("Unable to represent system_clock in GMT time");
         }
-        return PyDateTime_FromDateAndTime(localtime.tm_year + 1900,
-                                          localtime.tm_mon + 1,
-                                          localtime.tm_mday,
-                                          localtime.tm_hour,
-                                          localtime.tm_min,
-                                          localtime.tm_sec,
+        return PyDateTime_FromDateAndTime(gmtime.tm_year + 1900,
```

Comment:
@JamesUnicomb I am not sure about the UTC issue, but the datetime roundtrip issue can be easily resolved. We just need to use `PyDateTime_FromDateAndTimeAndFold` instead of `PyDateTime_FromDateAndTime` and make sure to properly set the `is_fold` argument.